src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java

```diff
@@ -3,30 +3,82 @@
 import core.basesyntax.dao.MessageDao;
 import core.basesyntax.model.Message;
 import java.util.List;
+import lombok.extern.log4j.Log4j;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
+@Log4j
 public class MessageDaoImpl extends AbstractDao implements MessageDao {
     public MessageDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Message create(Message entity) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                log.debug("Transaction creation for " + entity.toString()
+                        + " has been rollbacked.", e);
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Content entity", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        log.debug("Entity " + entity.toString() + " created");
+        return entity;
     }
 
     @Override
     public Message get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Message.class, id);
+        }
     }
 
     @Override
     public List<Message> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Message> getAllMessagesQuery =
+                    session.createQuery("from Message ", Message.class);
+            return getAllMessagesQuery.getResultList();
+        }
     }
 
     @Override
     public void remove(Message entity) {
-
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            Message message = session.find(Message.class, entity.getId());
+            if (message != null) {
```

Comment:
do we need this checking? better remove it

Reply:
Fixed